.ci/docker-setup.sh

```diff
@@ -8,56 +8,60 @@
 # - INTEGRATION: (optional) a value of "true" changes the default value of $TEST_MODE to "integration"
 set -e
 
+pull_docker_snapshot() {
+  project="${1?project name required}"
+  local docker_image="docker.elastic.co/${project}/${project}${DISTRIBUTION_SUFFIX}:${ELASTIC_STACK_VERSION}"
+  echo "Pulling $docker_image"
+  docker pull "$docker_image"
+}
+
 # TEST_MODE should be one of "unit" or "integration" (defaults to "unit" unless INTEGRATION=true)
 : "${TEST_MODE:=$([[ "${INTEGRATION}" = "true" ]] && echo "integration" || echo "unit")}"
 export TEST_MODE
 
- VERSION_URL="https://raw.githubusercontent.com/elastic/logstash/master/ci/logstash_releases.json"
+VERSION_URL="https://raw.githubusercontent.com/elastic/logstash/master/ci/logstash_releases.json"
 
- if [ -z "${ELASTIC_STACK_VERSION}" ]; then
+if [ -z "${ELASTIC_STACK_VERSION}" ]; then
     echo "Please set the ELASTIC_STACK_VERSION environment variable"
-    echo "For example: export ELASTIC_STACK_VERSION=6.2.4"
+    echo "For example: export ELASTIC_STACK_VERSION=7.x"
     exit 1
 fi
 
- echo "Fetching versions from $VERSION_URL"
+echo "Fetching versions from $VERSION_URL"
 VERSIONS=$(curl $VERSION_URL)
 
- if [[ "$SNAPSHOT" = "true" ]]; then
+if [[ "$SNAPSHOT" = "true" ]]; then
   ELASTIC_STACK_RETRIEVED_VERSION=$(echo $VERSIONS | jq '.snapshots."'"$ELASTIC_STACK_VERSION"'"')
   echo $ELASTIC_STACK_RETRIEVED_VERSION
 else
   ELASTIC_STACK_RETRIEVED_VERSION=$(echo $VERSIONS | jq '.releases."'"$ELASTIC_STACK_VERSION"'"')
 fi
 
- if [[ "$ELASTIC_STACK_RETRIEVED_VERSION" != "null" ]]; then
+if [[ "$ELASTIC_STACK_RETRIEVED_VERSION" != "null" ]]; then
   # remove starting and trailing double quotes
   ELASTIC_STACK_RETRIEVED_VERSION="${ELASTIC_STACK_RETRIEVED_VERSION%\"}"
   ELASTIC_STACK_RETRIEVED_VERSION="${ELASTIC_STACK_RETRIEVED_VERSION#\"}"
   echo "Translated $ELASTIC_STACK_VERSION to ${ELASTIC_STACK_RETRIEVED_VERSION}"
   export ELASTIC_STACK_VERSION=$ELASTIC_STACK_RETRIEVED_VERSION
 fi
 
- echo "Testing against version: $ELASTIC_STACK_VERSION"
-
- if [[ "$ELASTIC_STACK_VERSION" = *"-SNAPSHOT" ]]; then
```

Comment:
I've not a such a deep knowledge of docker, but `docker pull` is the same as this `curl` + `docker load` ?

Reply:
`docker pull` pull the image from the registry while `docker load` import image from STDIN .tar.gz